recod_frappe_devtools/build_docs/setup_docs.py

```diff
@@ -0,0 +1,368 @@
+"""Automatically setup docs for a project
+
+Call from command line:
+
+	bench setup-docs app path
+
+"""
+from __future__ import unicode_literals, print_function
+
+import os, json, frappe, shutil
+
+import jinja2
+from frappe.utils import markdown
+
+
+# def is_singe_doc(basepath, doctype):
+#     if "__" not in doctype:
+#         with open(basepath + "/" + doctype + ".json", 'r') as f:
+#             data_str = f.read()
+#             data = json.loads(data_str)
+#             return data['issingle'] == 1
+#     return True
+
+
+class SetupDocs(object):
+    def __init__(self, app, target_app):
+
+        self.list_sidebar = []
+        self.app = app
+        self.target_app = target_app
+
+        frappe.flags.web_pages_folders = ['docs', ]
+        frappe.flags.web_pages_apps = [self.app, ]
+        self.list_without_app_documentation = ['frappe', 'erpnext']
+
+        self.hooks = frappe.get_hooks(app_name=self.app)
+        self.app_title = self.hooks.get("app_title")[0]
+        self.setup_app_context()
+
+    def setup_app_context(self):
+        self.docs_config = frappe.get_module(self.app + ".config.docs")
+        version = get_version(app=self.app)
+        self.app_context = {
+            "app": frappe._dict({
+                "name": self.app,
+                "title": self.app_title,
+                "description": self.hooks.get("app_description")[0],
+                "version": version,
+                "publisher": self.hooks.get("app_publisher")[0],
+                "icon": self.hooks.get("app_icon")[0],
+                "email": self.hooks.get("app_email")[0],
+                "source_link": self.docs_config.source_link,
+                "license": self.hooks.get("app_license")[0],
+                "branch": getattr(self.docs_config, "branch", None) or "develop",
+            }),
+            "metatags": {
+                "description": self.hooks.get("app_description")[0],
+            },
+            "get_doctype_app": frappe.get_doctype_app
+        }
+
+    def get_raw_for_md_file(self, app_title, app_name):
+        if not os.path.exists(frappe.get_app_path(app_name, 'www')):
+            self.list_without_app_documentation.append(app_name)
```

Comment:
Bad idea. The `www` directory in an app is not only used for documentation but for any website content, so you cannot _assume_ that if the directory exists there is a documentation.

The good solution would be to generate a JSON or txt file in docs home with the list of apps documentation. Everytime the `bench build-app-docs` command is called, the _id_ of the app should be added to the file (if it does not already exists).

Example workflow:
1. _docs_apps.txt_:
```
```
2. `bench build-app-docs myapp`
3. _docs_apps.txt_:
```
myapp
```

Read the content of this file to know the list of apps that generated a documentation.